### PR TITLE
Document breaking change in data source aws_launch_template 

### LINKED ISSUE #15094: Upgrade to 3.0 documentation does not indicate breaking change in aws_launch_template data source
According to https://github.com/terraform-providers/terraform-provider-aws/issues/13410 there is a behaviour change in the data source so that if no match is found an error is raised.  This change is not documented in https://registry.terraform.io/providers/hashicorp/aws/latest/docs/guides/version-3-upgrade.

As a side note, the change does make sense but it should have been accompanied by having a data source like `aws_launch_templates` that can return an empty list.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v0.13.1
+ provider registry.terraform.io/hashicorp/aws v3.5.0

### Affected Resource(s)

* aws_launch_template

### References

* #13410

### PR DESCRIPTION
**Pull request by arielpeltz**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #15094 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
Document data source aws_launch_template breaking change in upgrade to 3.0
```

### CODE REVIEW

File: website/docs/guides/version-3-upgrade.html.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ Upgrade topics:
 - [Data Source: aws_lambda_invocation](#data-source-aws_lambda_invocation)
 - [Data Source: aws_route53_resolver_rule](#data-source-aws_route53_resolver_rule)
 - [Data Source: aws_route53_zone](#data-source-aws_route53_zone)
+- [Data Source: aws_launch_template](#data-source-aws_launch_template)
```
Comment (anGie44) [inline website/docs/guides/version-3-upgrade.html.md:28]: Hi @arielpeltz 👋 Do you mind moving this line below `aws_lambda` to maintain the list in alphabetical order?

File: website/docs/guides/version-3-upgrade.html.md [hunk 2]
```diff
@@ -231,6 +232,21 @@ While the returned value will omit the trailing period, use of configurations wi
 Previously the data-source returned the Hosted Zone Domain Name directly from the API, which included a `.` suffix. This proves difficult when many other AWS services do not accept this trailing period (e.g. ACM Certificate). This period is now automatically removed. For example, when the attribute would previously return a Hosted Zone Domain Name such as `example.com.`, the attribute now will be returned as `example.com`.
 While the returned value will omit the trailing period, use of configurations with trailing periods will not be interrupted.
 
+## Data Source: aws_launch_template
```
Comment (anGie44) [inline website/docs/guides/version-3-upgrade.html.md:235]: Nit: similar comment as above to move this section

**Review by anGie44 — CHANGES_REQUESTED** [review]
Thank you for this documentation update @arielpeltz! Just a minor comment regarding the physical ordering of the the items, otherwise looks great!

**Review by anGie44 — APPROVED** [review]
Thanks @arielpeltz, LGTM 👍

**Comment by arielpeltz** [comment]
That was quick @anGie44 .  I was waiting for the tests to finish before requesting a review...  Thanks!!

**Pull request merged**